satpy/readers/hsaf_h5.py

```diff
@@ -0,0 +1,135 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+# Copyright (c) 2019.
+#
+# This file is part of satpy.
+#
+# satpy is free software: you can redistribute it and/or modify it under the
+# terms of the GNU General Public License as published by the Free Software
+# Foundation, either version 3 of the License, or (at your option) any later
+# version.
+#
+# satpy is distributed in the hope that it will be useful, but WITHOUT ANY
+# WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR
+# A PARTICULAR PURPOSE.  See the GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License along with
+# satpy.  If not, see <http://www.gnu.org/licenses/>.
+"""A reader for HDF5 Snow Cover (SC) file produced by the Hydrology SAF."""
+import logging
+from datetime import timedelta
+
+import dask.array as da
+import h5py
+import numpy as np
+import xarray as xr
+
+from satpy import CHUNK_SIZE
+from satpy.readers.file_handlers import BaseFileHandler
+from satpy.resample import get_area_def
+
+LOG = logging.getLogger(__name__)
+AREA_X_OFFSET = 1211
+AREA_Y_OFFSET = 62
+
+
+class HSAFFileHandler(BaseFileHandler):
+    """File handler for HSAF H5 files."""
+
+    def __init__(self, filename, filename_info, filetype_info):
+        """Init the file handler."""
+        super(HSAFFileHandler, self).__init__(filename,
+                                              filename_info,
+                                              filetype_info)
+        self._h5fh = h5py.File(self.filename, 'r')
+
+    @property
+    def end_time(self):
+        """Get end time."""
+        return self.start_time + timedelta(hours=23, minutes=59, seconds=59)
+
+    @property
+    def start_time(self):
+        """Get start time."""
+        return self.filename_info['sensing_time']
+
+    def _prepare_variable_for_palette(self, dset, ds_info):
+        colormap = np.array(np.array(dset))
```

Comment:
The outer `np.array()` is redundant.
```suggestion
        colormap = np.array(dset)
```